proto/cosmos/upgrade/v1beta1/upgrade.proto

```diff
@@ -40,6 +40,39 @@ message Plan {
   // If this field is not empty, an error will be thrown.
   google.protobuf.Any upgraded_client_state = 5
       [deprecated = true, (gogoproto.moretags) = "yaml:\"upgraded_client_state\""];
+
+  // Optional: Upgrade contains additional instructions for the devops or a hypervisor.
+  // App specific instructions are handled by the `info` attribute. Here we provide
+  // information such as pre-upgrade or post-upgrade commands.
+  UpgradeInstructions Upgrade = 6;
+}
+
+message UpgradeInstructions {
+  option (gogoproto.equal)            = true;
+
+  // If not empty, a shell command to be run by the upgrade manager or a hypervisor after shutting down
+  // the app and before running a new node.
+  string pre_run = 1;
+  // If not empty, a shell command to be run by the upgrade manager or a hypervisor after shutting down
+  // the app and after running a new app.
+  string post_run = 2;
```

Comment:
Do you know any use case for this? Also how would you deal with the errors on the post upgrade? The application is already running in theory but your post upgrade script fails. What would you do?

Reply:
One possible use-case is reporting.
But maybe it's not needed. I'm happy to remove it if you think it may confuse people or if it will be abused.

Reply:
I believe this post-run script can be used for non-critical activities, so we don't need to handle errors. We can use it for removing backups, alerting users etc